pkg/operator/admission/webhook.go

```diff
@@ -0,0 +1,203 @@
+	if err != nil {
+		// If secret is not found. All good ! Proceed with rook without admission controllers
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func createWebhookService(context *clusterd.Context) error {
+	webhookService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: servicePort,
+					TargetPort: intstr.IntOrString{
+						IntVal: serverPort,
+					},
+				},
+			},
+
+			Selector: map[string]string{
+				"app": "rook-ceph-admission-controller",
+			},
+		},
+	}
+	_, err := context.Clientset.CoreV1().Services(namespace).Create(&webhookService)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// InitWebhookIfSecretPresent will initialize the webhook if secret is detected
+func InitWebhookIfSecretPresent(context *clusterd.Context, admissionImage string) error {
+	isPresent, err := isSecretPresent(context)
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve secret")
+	}
+	if isPresent {
+		err = initWebhook(context, admissionImage)
+		if err != nil {
+			return errors.Wrap(err, "failed to initialize webhook")
+		}
+	}
+	return nil
+}
+
+func initWebhook(context *clusterd.Context, admissionImage string) error {
+	// At this point volume should be mounted, so proceed with creating the service and validatingwebhookconfig
+	err := createWebhookService(context)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service")
+	}
+	err = createWebhookPods(context, admissionImage)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service")
+	}
+	return nil
+}
+
+func createWebhookPods(context *clusterd.Context, admissionImage string) error {
+	admission_parameters := []string{"admission-controller",
+		"--storage", storageFlag}
+	secretVolume := getSecretVolume()
+	secretVolumeMount := getSecretVolumeMount()
+
+	antiAffinity := csi.GetPodAntiAffinity("app", "rook-ceph-admission-controller")
+	admissionControllerDeployment := getDeployment(secretVolume, antiAffinity, admissionImage, admission_parameters, secretVolumeMount)
+
+	_, err := context.Clientset.AppsV1().Deployments(namespace).Create(&admissionControllerDeployment)
+	if err != nil {
+		return errors.Wrap(err, "failed to create admission controller replica set")
+	}
+
+	return nil
+}
+
+func getDeployment(secretVolume corev1.Volume, antiAffinity corev1.PodAntiAffinity, admissionImage string, admission_parameters []string, secretVolumeMount corev1.VolumeMount) v1.Deployment {
+	var replicas int32 = 2
+	admissionControllerDeployment := v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: namespace,
+		},
+		Spec: v1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "rook-ceph-admission-controller",
```

Comment:
please make `rook-ceph-admission-controller` a constant `appName`

Reply:
Additionally, `app` is defined in `pkg/operator/k8sutil/pod.go`: `AppAttr`.

Reply:
Done